File: backend/core/src/modules/ai-requests/ai-requests.controller.ts
import { Controller, UseGuards } from '@nestjs/common';
import { UserScopedCrudController } from '../../common/access/user-scoped-crud.controller';
import { AiRequestsService } from './ai-requests.service';
import { AiRequest } from '../../entities';
import { JwtAuthGuard } from '../auth/jwt-auth.guard';

@UseGuards(JwtAuthGuard)
@Controller('ai-requests')
export class AiRequestsController extends UserScopedCrudController<AiRequest> {
  constructor(service: AiRequestsService) {
    super(service);
  }
}

File: backend/core/src/main/java/store/pjcloud/cowrite/core/controller/AiRequestsController.java
package store.pjcloud.cowrite.core.controller;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import store.pjcloud.cowrite.core.common.UserScopedCrudController;
import store.pjcloud.cowrite.core.entity.AiRequest;
import store.pjcloud.cowrite.core.service.AiRequestsService;

@RestController
@RequestMapping("/ai-requests")
public class AiRequestsController extends UserScopedCrudController<AiRequest> {
    public AiRequestsController(AiRequestsService service) {
        super(service);
    }
}
